snitch doesn't compile

context: 1.2.2 release, header-only mode, newest VS (17.8.0 preview 2)

Even compiling a trivial

```c++
#define SNITCH_IMPLEMENTATION
#include <snitch/snitch_all.hpp>

TEST_CASE("Factorials are computed", "[factorial]") {
   REQUIRE(true);
}
```

results in `error C2397: conversion from 'long' to 'size_t' requires a narrowing conversion`

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,6 +1,6 @@
 cmake_minimum_required(VERSION 3.14)
 
-project(snitch LANGUAGES CXX VERSION 1.2.2)
+project(snitch LANGUAGES CXX VERSION 1.2.3)
 
 # Maximum lengths.
 set(SNITCH_MAX_TEST_CASES           5000 CACHE STRING "Maximum number of test cases in a test application.")

--- a/include/snitch/snitch_macros_check.hpp
+++ b/include/snitch/snitch_macros_check.hpp
@@ -37,14 +37,14 @@
     do {                                                                                           \
         auto& SNITCH_CURRENT_TEST = snitch::impl::get_current_test();                              \
         SNITCH_CURRENT_TEST.reg.report_assertion(                                                  \
-            true, SNITCH_CURRENT_TEST, {__FILE__, __LINE__}, (MESSAGE));                           \
+            true, SNITCH_CURRENT_TEST, SNITCH_CURRENT_LOCATION, (MESSAGE));                        \
     } while (0)
 
 #define SNITCH_FAIL(MESSAGE)                                                                       \
     do {                                                                                           \
         auto& SNITCH_CURRENT_TEST = snitch::impl::get_current_test();                              \
         SNITCH_CURRENT_TEST.reg.report_assertion(                                                  \
-            false, SNITCH_CURRENT_TEST, {__FILE__, __LINE__}, (MESSAGE));                          \
+            false, SNITCH_CURRENT_TEST, SNITCH_CURRENT_LOCATION, (MESSAGE));                       \
         SNITCH_TESTING_ABORT;                                                                      \
     } while (0)
 
@@ -52,14 +52,14 @@
     do {                                                                                           \
         auto& SNITCH_CURRENT_TEST = snitch::impl::get_current_test();                              \
         SNITCH_CURRENT_TEST.reg.report_assertion(                                                  \
-            false, SNITCH_CURRENT_TEST, {__FILE__, __LINE__}, (MESSAGE));                          \
+            false, SNITCH_CURRENT_TEST, SNITCH_CURRENT_LOCATION, (MESSAGE));                       \
     } while (0)
 
 #define SNITCH_SKIP(MESSAGE)                                                                       \
     do {                                                                                           \
         auto& SNITCH_CURRENT_TEST = snitch::impl::get_current_test();                              \
         SNITCH_CURRENT_TEST.reg.report_skipped(                                                    \
-            SNITCH_CURRENT_TEST, {__FILE__, __LINE__}, (MESSAGE));                                 \
+            SNITCH_CURRENT_TEST, SNITCH_CURRENT_LOCATION, (MESSAGE));                              \
         SNITCH_TESTING_ABORT;                                                                      \
     } while (0)
 
@@ -67,7 +67,7 @@
     do {                                                                                           \
         auto& SNITCH_CURRENT_TEST = snitch::impl::get_current_test();                              \
         SNITCH_CURRENT_TEST.reg.report_skipped(                                                    \
-            SNITCH_CURRENT_TEST, {__FILE__, __LINE__}, (MESSAGE));                                 \
+            SNITCH_CURRENT_TEST, SNITCH_CURRENT_LOCATION, (MESSAGE));                              \
     } while (0)
 
 #define SNITCH_REQUIRE_THAT_IMPL(CHECK, MAYBE_ABORT, EXPR, ...)                                    \

--- a/include/snitch/snitch_macros_check_base.hpp
+++ b/include/snitch/snitch_macros_check_base.hpp
@@ -26,7 +26,7 @@
 
 #define SNITCH_REPORT_EXPRESSION(MAYBE_ABORT)                                                      \
     SNITCH_CURRENT_TEST.reg.report_assertion(                                                      \
-        SNITCH_CURRENT_EXPRESSION.success, SNITCH_CURRENT_TEST, {__FILE__, __LINE__},              \
+        SNITCH_CURRENT_EXPRESSION.success, SNITCH_CURRENT_TEST, SNITCH_CURRENT_LOCATION,           \
         SNITCH_CURRENT_EXPRESSION);                                                                \
     if (!SNITCH_CURRENT_EXPRESSION.success) {                                                      \
         MAYBE_ABORT;                                                                               \

--- a/include/snitch/snitch_macros_exceptions.hpp
+++ b/include/snitch/snitch_macros_exceptions.hpp
@@ -14,25 +14,25 @@
             try {                                                                                  \
                 static_cast<void>(EXPRESSION);                                                     \
                 SNITCH_CURRENT_TEST.reg.report_assertion(                                          \
-                    false, SNITCH_CURRENT_TEST, {__FILE__, __LINE__},                              \
+                    false, SNITCH_CURRENT_TEST, SNITCH_CURRENT_LOCATION,                           \
                     #__VA_ARGS__ " expected but no exception thrown");                             \
                 MAYBE_ABORT;                                                                       \
             } catch (const __VA_ARGS__&) {                                                         \
                 SNITCH_CURRENT_TEST.reg.report_assertion(                                          \
-                    true, SNITCH_CURRENT_TEST, {__FILE__, __LINE__},                               \
+                    true, SNITCH_CURRENT_TEST, SNITCH_CURRENT_LOCATION,                            \
                     #__VA_ARGS__ " was thrown as expected");                                       \
             } catch (...) {                                                                        \
                 try {                                                                              \
                     throw;                                                                         \
                 } catch (const std::exception& e) {                                                \
                     SNITCH_CURRENT_TEST.reg.report_assertion(                                      \
-                        false, SNITCH_CURRENT_TEST, {__FILE__, __LINE__},                          \
+                        false, SNITCH_CURRENT_TEST, SNITCH_CURRENT_LOCATION,                       \
                         #__VA_ARGS__ " expected but other std::exception thrown; message: ",       \
                         e.what());                                                                 \
                     MAYBE_ABORT;                                                                   \
                 } catch (...) {                                                                    \
                     SNITCH_CURRENT_TEST.reg.report_assertion(                                      \
-                        false, SNITCH_CURRENT_TEST, {__FILE__, __LINE__},                          \
+                        false, SNITCH_CURRENT_TEST, SNITCH_CURRENT_LOCATION,                       \
                         #__VA_ARGS__ " expected but other unknown exception thrown");              \
                     MAYBE_ABORT;                                                                   \
                 }                                                                                  \
@@ -50,21 +50,21 @@
             try {                                                                                  \
                 static_cast<void>(EXPRESSION);                                                     \
                 SNITCH_CURRENT_TEST.reg.report_assertion(                                          \
-                    false, SNITCH_CURRENT_TEST, {__FILE__, __LINE__},                              \
+                    false, SNITCH_CURRENT_TEST, SNITCH_CURRENT_LOCATION,                           \
                     #EXCEPTION " expected but no exception thrown");                               \
                 MAYBE_ABORT;                                                                       \
             } catch (const EXCEPTION& e) {                                                         \
                 auto&& SNITCH_TEMP_MATCHER = __VA_ARGS__;                                          \
                 if (!SNITCH_TEMP_MATCHER.match(e)) {                                               \
                     SNITCH_CURRENT_TEST.reg.report_assertion(                                      \
-                        false, SNITCH_CURRENT_TEST, {__FILE__, __LINE__},                          \
+                        false, SNITCH_CURRENT_TEST, SNITCH_CURRENT_LOCATION,                       \
                         "could not match caught " #EXCEPTION " with expected content: ",           \
                         SNITCH_TEMP_MATCHER.describe_match(                                        \
                             e, snitch::matchers::match_status::failed));                           \
                     MAYBE_ABORT;                                                                   \
                 } else {                                                                           \
                     SNITCH_CURRENT_TEST.reg.report_assertion(                                      \
-                        true, SNITCH_CURRENT_TEST, {__FILE__, __LINE__},                           \
+                        true, SNITCH_CURRENT_TEST, SNITCH_CURRENT_LOCATION,                        \
                         "caught " #EXCEPTION " matched expected content: ",                        \
                         SNITCH_TEMP_MATCHER.describe_match(                                        \
                             e, snitch::matchers::match_status::matched));                          \
@@ -74,13 +74,13 @@
                     throw;                                                                         \
                 } catch (const std::exception& e) {                                                \
                     SNITCH_CURRENT_TEST.reg.report_assertion(                                      \
-                        false, SNITCH_CURRENT_TEST, {__FILE__, __LINE__},                          \
+                        false, SNITCH_CURRENT_TEST, SNITCH_CURRENT_LOCATION,                       \
                         #EXCEPTION " expected but other std::exception thrown; message: ",         \
                         e.what());                                                                 \
                     MAYBE_ABORT;                                                                   \
                 } catch (...) {                                                                    \
                     SNITCH_CURRENT_TEST.reg.report_assertion(                                      \
-                        false, SNITCH_CURRENT_TEST, {__FILE__, __LINE__},                          \
+                        false, SNITCH_CURRENT_TEST, SNITCH_CURRENT_LOCATION,                       \
                         #EXCEPTION " expected but other unknown exception thrown");                \
                     MAYBE_ABORT;                                                                   \
                 }                                                                                  \
@@ -98,20 +98,20 @@
             try {                                                                                  \
                 static_cast<void>(__VA_ARGS__);                                                    \
                 SNITCH_CURRENT_TEST.reg.report_assertion(                                          \
-                    true, SNITCH_CURRENT_TEST, {__FILE__, __LINE__},                               \
+                    true, SNITCH_CURRENT_TEST, SNITCH_CURRENT_LOCATION,                            \
                     #__VA_ARGS__ " did not throw");                                                \
             } catch (...) {                                                                        \
                 try {                                                                              \
                     throw;                                                                         \
                 } catch (const std::exception& e) {                                                \
                     SNITCH_CURRENT_TEST.reg.report_assertion(                                      \
-                        false, SNITCH_CURRENT_TEST, {__FILE__, __LINE__},                          \
+                        false, SNITCH_CURRENT_TEST, SNITCH_CURRENT_LOCATION,                       \
                         "expected " #__VA_ARGS__                                                   \
                         " not to throw but it threw a std::exception; message: ",                  \
                         e.what());                                                                 \
                 } catch (...) {                                                                    \
                     SNITCH_CURRENT_TEST.reg.report_assertion(                                      \
-                        false, SNITCH_CURRENT_TEST, {__FILE__, __LINE__},                          \
+                        false, SNITCH_CURRENT_TEST, SNITCH_CURRENT_LOCATION,                       \
                         "expected " #__VA_ARGS__                                                   \
                         " not to throw but it threw an unknown exception");                        \
                 }                                                                                  \

--- a/include/snitch/snitch_macros_misc.hpp
+++ b/include/snitch/snitch_macros_misc.hpp
@@ -9,7 +9,7 @@
 
 #define SNITCH_SECTION(...)                                                                        \
     if (snitch::impl::section_entry_checker SNITCH_MACRO_CONCAT(section_id_, __COUNTER__){         \
-            {{__VA_ARGS__}, {__FILE__, __LINE__}}, snitch::impl::get_current_test()})
+            {{__VA_ARGS__}, SNITCH_CURRENT_LOCATION}, snitch::impl::get_current_test()})
 
 #define SNITCH_CAPTURE(...)                                                                        \
     auto SNITCH_MACRO_CONCAT(capture_id_, __COUNTER__) =                                           \

--- a/include/snitch/snitch_macros_test_case.hpp
+++ b/include/snitch/snitch_macros_test_case.hpp
@@ -8,7 +8,7 @@
 #define SNITCH_TEST_CASE_IMPL(ID, ...)                                                             \
     static void        ID();                                                                       \
     static const char* SNITCH_MACRO_CONCAT(test_id_, __COUNTER__) [[maybe_unused]] =               \
-        snitch::tests.add({__VA_ARGS__}, {__FILE__, __LINE__}, &ID);                               \
+        snitch::tests.add({__VA_ARGS__}, SNITCH_CURRENT_LOCATION, &ID);                            \
     void ID()
 
 #define SNITCH_TEST_CASE(...)                                                                      \
@@ -19,7 +19,7 @@
     static void        ID();                                                                       \
     static const char* SNITCH_MACRO_CONCAT(test_id_, __COUNTER__) [[maybe_unused]] =               \
         snitch::tests.add_with_type_list<TYPES>(                                                   \
-            {NAME, TAGS}, {__FILE__, __LINE__}, []<typename TestType>() { ID<TestType>(); });      \
+            {NAME, TAGS}, SNITCH_CURRENT_LOCATION, []<typename TestType>() { ID<TestType>(); });   \
     template<typename TestType>                                                                    \
     void ID()
 
@@ -32,7 +32,7 @@
     static void        ID();                                                                       \
     static const char* SNITCH_MACRO_CONCAT(test_id_, __COUNTER__) [[maybe_unused]] =               \
         snitch::tests.add_with_types<__VA_ARGS__>(                                                 \
-            {NAME, TAGS}, {__FILE__, __LINE__}, []<typename TestType>() { ID<TestType>(); });      \
+            {NAME, TAGS}, SNITCH_CURRENT_LOCATION, []<typename TestType>() { ID<TestType>(); });   \
     template<typename TestType>                                                                    \
     void ID()
 
@@ -48,7 +48,7 @@
     }                                                                                              \
     static const char* SNITCH_MACRO_CONCAT(test_id_, __COUNTER__) [[maybe_unused]] =               \
         snitch::tests.add_fixture(                                                                 \
-            {#FIXTURE, __VA_ARGS__}, {__FILE__, __LINE__}, []() { ID{}.test_fun(); });             \
+            {#FIXTURE, __VA_ARGS__}, SNITCH_CURRENT_LOCATION, []() { ID{}.test_fun(); });          \
     void ID::test_fun()
 
 #define SNITCH_TEST_CASE_METHOD(FIXTURE, ...)                                                      \
@@ -64,7 +64,7 @@
     }                                                                                              \
     static const char* SNITCH_MACRO_CONCAT(test_id_, __COUNTER__) [[maybe_unused]] =               \
         snitch::tests.add_fixture_with_type_list<TYPES>(                                           \
-            {#FIXTURE, NAME, TAGS}, {__FILE__, __LINE__},                                          \
+            {#FIXTURE, NAME, TAGS}, SNITCH_CURRENT_LOCATION,                                       \
             []() < typename TestType > { ID<TestType>{}.test_fun(); });                            \
     template<typename TestType>                                                                    \
     void ID<TestType>::test_fun()
@@ -82,7 +82,7 @@
     }                                                                                              \
     static const char* SNITCH_MACRO_CONCAT(test_id_, __COUNTER__) [[maybe_unused]] =               \
         snitch::tests.add_fixture_with_types<__VA_ARGS__>(                                         \
-            {#FIXTURE, NAME, TAGS}, {__FILE__, __LINE__},                                          \
+            {#FIXTURE, NAME, TAGS}, SNITCH_CURRENT_LOCATION,                                       \
             []() < typename TestType > { ID<TestType>{}.test_fun(); });                            \
     template<typename TestType>                                                                    \
     void ID<TestType>::test_fun()

--- a/include/snitch/snitch_macros_utility.hpp
+++ b/include/snitch/snitch_macros_utility.hpp
@@ -6,4 +6,7 @@
 #define SNITCH_CONCAT_IMPL(x, y) x##y
 #define SNITCH_MACRO_CONCAT(x, y) SNITCH_CONCAT_IMPL(x, y)
 
+#define SNITCH_CURRENT_LOCATION                                                                    \
+    snitch::source_location { std::string_view{__FILE__}, static_cast<std::size_t>(__LINE__) }
+
 #endif

--- a/meson.build
+++ b/meson.build
@@ -1,6 +1,6 @@
 project('snitch', 'cpp',
   default_options: ['cpp_std=c++20', 'default_library=static'],
-  version: '1.2.2'
+  version: '1.2.3'
 )
 
 include_dirs = include_directories('.', 'include')

--- a/tests/approval_tests/reporter_catch2_xml.cpp
+++ b/tests/approval_tests/reporter_catch2_xml.cpp
@@ -11,10 +11,10 @@ using snitch::matchers::contains_substring;
 TEST_CASE("teamcity reporter", "[reporters]") {
     mock_framework framework;
     register_tests_for_reporters(framework.registry);
-    framework.registry.add({"test escape <>&\"'"}, {__FILE__, __LINE__}, [] {
+    framework.registry.add({"test escape <>&\"'"}, SNITCH_CURRENT_LOCATION, [] {
         SNITCH_FAIL("escape <>&\"' in messages");
     });
-    framework.registry.add({"test escape very long"}, {__FILE__, __LINE__}, [] {
+    framework.registry.add({"test escape very long"}, SNITCH_CURRENT_LOCATION, [] {
         SNITCH_FAIL(std::string(2 * snitch::max_message_length, '&'));
     });
 

--- a/tests/approval_tests/reporter_teamcity.cpp
+++ b/tests/approval_tests/reporter_teamcity.cpp
@@ -11,10 +11,10 @@ using snitch::matchers::contains_substring;
 TEST_CASE("teamcity reporter", "[reporters]") {
     mock_framework framework;
     register_tests_for_reporters(framework.registry);
-    framework.registry.add({"test escape |'\n\r[]"}, {__FILE__, __LINE__}, [] {
+    framework.registry.add({"test escape |'\n\r[]"}, SNITCH_CURRENT_LOCATION, [] {
         SNITCH_FAIL("escape | message || | '\n\r[]");
     });
-    framework.registry.add({"test escape very long"}, {__FILE__, __LINE__}, [] {
+    framework.registry.add({"test escape very long"}, SNITCH_CURRENT_LOCATION, [] {
         SNITCH_FAIL(std::string(2 * snitch::max_message_length, '|'));
     });
 

--- a/tests/runtime_tests/registry.cpp
+++ b/tests/runtime_tests/registry.cpp
@@ -24,7 +24,7 @@ TEST_CASE("add regular test", "[registry]") {
 
     test_called = false;
     framework.registry.add(
-        {"how many lights", "[tag]"}, {__FILE__, __LINE__}, []() { test_called = true; });
+        {"how many lights", "[tag]"}, SNITCH_CURRENT_LOCATION, []() { test_called = true; });
 
     REQUIRE(framework.get_num_registered_tests() == 1u);
 
@@ -49,7 +49,8 @@ TEST_CASE("add regular test no tags", "[registry]") {
     mock_framework framework;
 
     test_called = false;
-    framework.registry.add({"how many lights"}, {__FILE__, __LINE__}, []() { test_called = true; });
+    framework.registry.add(
+        {"how many lights"}, SNITCH_CURRENT_LOCATION, []() { test_called = true; });
 
     REQUIRE(framework.get_num_registered_tests() == 1u);
 
@@ -82,7 +83,7 @@ TEST_CASE("add template test", "[registry]") {
 
         if (with_type_list) {
             framework.registry.add_with_type_list<snitch::type_list<int, float>>(
-                {"how many lights", "[tag]"}, {__FILE__, __LINE__}, []<typename T>() {
+                {"how many lights", "[tag]"}, SNITCH_CURRENT_LOCATION, []<typename T>() {
                     if constexpr (std::is_same_v<T, int>) {
                         test_called_int = true;
                     } else if constexpr (std::is_same_v<T, float>) {
@@ -93,7 +94,7 @@ TEST_CASE("add template test", "[registry]") {
                 });
         } else {
             framework.registry.add_with_types<int, float>(
-                {"how many lights", "[tag]"}, {__FILE__, __LINE__}, []<typename T>() {
+                {"how many lights", "[tag]"}, SNITCH_CURRENT_LOCATION, []<typename T>() {
                     if constexpr (std::is_same_v<T, int>) {
                         test_called_int = true;
                     } else if constexpr (std::is_same_v<T, float>) {
@@ -170,7 +171,7 @@ void finish(snitch::registry&) noexcept {
 }
 
 void register_one_test(snitch::registry& r) {
-    r.add({"the test", "[tag]"}, {__FILE__, __LINE__}, []() { SNITCH_CHECK(1 == 2); });
+    r.add({"the test", "[tag]"}, SNITCH_CURRENT_LOCATION, []() { SNITCH_CHECK(1 == 2); });
 }
 }} // namespace ::my_reporter
 
@@ -307,20 +308,20 @@ void register_tests(mock_framework& framework) {
     test_called_hidden2   = false;
 
     framework.registry.add(
-        {"how are you", "[tag]"}, {__FILE__, __LINE__}, []() { test_called = true; });
+        {"how are you", "[tag]"}, SNITCH_CURRENT_LOCATION, []() { test_called = true; });
 
-    framework.registry.add({"how many lights", "[tag][other_tag]"}, {__FILE__, __LINE__}, []() {
+    framework.registry.add({"how many lights", "[tag][other_tag]"}, SNITCH_CURRENT_LOCATION, []() {
         test_called_other_tag = true;
         SNITCH_FAIL_CHECK("there are four lights");
     });
 
-    framework.registry.add({"drink from the cup", "[tag][skipped]"}, {__FILE__, __LINE__}, []() {
+    framework.registry.add({"drink from the cup", "[tag][skipped]"}, SNITCH_CURRENT_LOCATION, []() {
         test_called_skipped = true;
         SNITCH_SKIP_CHECK("not thirsty");
     });
 
     framework.registry.add_with_types<int, float>(
-        {"how many templated lights", "[tag][tag with spaces]"}, {__FILE__, __LINE__},
+        {"how many templated lights", "[tag][tag with spaces]"}, SNITCH_CURRENT_LOCATION,
         []<typename T>() {
             if constexpr (std::is_same_v<T, int>) {
                 test_called_int = true;
@@ -331,35 +332,37 @@ void register_tests(mock_framework& framework) {
             }
         });
 
-    framework.registry.add({"hidden test 1", "[.][hidden][other_tag]"}, {__FILE__, __LINE__}, []() {
-        test_called_hidden1 = true;
+    framework.registry.add(
+        {"hidden test 1", "[.][hidden][other_tag]"}, SNITCH_CURRENT_LOCATION,
+        []() { test_called_hidden1 = true; });
+
+    framework.registry.add({"hidden test 2", "[.hidden]"}, SNITCH_CURRENT_LOCATION, []() {
+        test_called_hidden2 = true;
     });
 
     framework.registry.add(
-        {"hidden test 2", "[.hidden]"}, {__FILE__, __LINE__}, []() { test_called_hidden2 = true; });
-
-    framework.registry.add(
-        {"may fail that does not fail", "[.][may fail][!mayfail]"}, {__FILE__, __LINE__}, []() {});
-
-    framework.registry.add(
-        {"may fail that does fail", "[.][may fail][!mayfail]"}, {__FILE__, __LINE__},
-        []() { SNITCH_FAIL_CHECK("it did fail"); });
-
-    framework.registry.add(
-        {"should fail that does not fail", "[.][should fail][!shouldfail]"}, {__FILE__, __LINE__},
+        {"may fail that does not fail", "[.][may fail][!mayfail]"}, SNITCH_CURRENT_LOCATION,
         []() {});
 
     framework.registry.add(
-        {"should fail that does fail", "[.][should fail][!shouldfail]"}, {__FILE__, __LINE__},
+        {"may fail that does fail", "[.][may fail][!mayfail]"}, SNITCH_CURRENT_LOCATION,
+        []() { SNITCH_FAIL_CHECK("it did fail"); });
+
+    framework.registry.add(
+        {"should fail that does not fail", "[.][should fail][!shouldfail]"},
+        SNITCH_CURRENT_LOCATION, []() {});
+
+    framework.registry.add(
+        {"should fail that does fail", "[.][should fail][!shouldfail]"}, SNITCH_CURRENT_LOCATION,
         []() { SNITCH_FAIL_CHECK("it did fail"); });
 
     framework.registry.add(
         {"may+should fail that does not fail", "[.][may+should fail][!mayfail][!shouldfail]"},
-        {__FILE__, __LINE__}, []() {});
+        SNITCH_CURRENT_LOCATION, []() {});
 
     framework.registry.add(
         {"may+should fail that does fail", "[.][may+should fail][!mayfail][!shouldfail]"},
-        {__FILE__, __LINE__}, []() { SNITCH_FAIL_CHECK("it did fail"); });
+        SNITCH_CURRENT_LOCATION, []() { SNITCH_FAIL_CHECK("it did fail"); });
 }
 } // namespace
 

--- a/tests/testing_reporters.cpp
+++ b/tests/testing_reporters.cpp
@@ -19,84 +19,87 @@ void register_tests_for_reporters(snitch::registry& r) {
     // To avoid unnecessary changes to the approval test data, please add new tests only at the
     // end of this list.
 
-    r.add({"test pass", "[tag2][tag1]"}, {__FILE__, __LINE__}, []() {});
-    r.add({"test fail", "[tag2][tag1]"}, {__FILE__, __LINE__}, []() { SNITCH_CHECK(1 == 2); });
+    r.add({"test pass", "[tag2][tag1]"}, SNITCH_CURRENT_LOCATION, []() {});
+    r.add({"test fail", "[tag2][tag1]"}, SNITCH_CURRENT_LOCATION, []() { SNITCH_CHECK(1 == 2); });
 
-    r.add({"test mayfail good pass", "[tag2][tag1][!mayfail]"}, {__FILE__, __LINE__}, []() {});
-    r.add({"test mayfail bad pass", "[tag2][tag1][!mayfail]"}, {__FILE__, __LINE__}, []() {
+    r.add({"test mayfail good pass", "[tag2][tag1][!mayfail]"}, SNITCH_CURRENT_LOCATION, []() {});
+    r.add({"test mayfail bad pass", "[tag2][tag1][!mayfail]"}, SNITCH_CURRENT_LOCATION, []() {
         SNITCH_CHECK(1 == 2);
     });
     r.add(
-        {"test shouldfail good fail", "[tag2][tag1][!shouldfail]"}, {__FILE__, __LINE__}, []() {});
-    r.add({"test shouldfail bad pass", "[tag2][tag1][!shouldfail]"}, {__FILE__, __LINE__}, []() {
+        {"test shouldfail good fail", "[tag2][tag1][!shouldfail]"}, SNITCH_CURRENT_LOCATION,
+        []() {});
+    r.add({"test shouldfail bad pass", "[tag2][tag1][!shouldfail]"}, SNITCH_CURRENT_LOCATION, []() {
         SNITCH_CHECK(1 == 2);
     });
 
-    r.add({"test no tags pass"}, {__FILE__, __LINE__}, []() {});
-    r.add({"test no tags fail"}, {__FILE__, __LINE__}, []() { SNITCH_CHECK(1 == 2); });
+    r.add({"test no tags pass"}, SNITCH_CURRENT_LOCATION, []() {});
+    r.add({"test no tags fail"}, SNITCH_CURRENT_LOCATION, []() { SNITCH_CHECK(1 == 2); });
 
     r.add_with_types<int, float>(
-        {"typed test no tags pass"}, {__FILE__, __LINE__}, []<typename T>() {});
+        {"typed test no tags pass"}, SNITCH_CURRENT_LOCATION, []<typename T>() {});
     r.add_with_types<int, float>(
-        {"typed test no tags fail"}, {__FILE__, __LINE__},
+        {"typed test no tags fail"}, SNITCH_CURRENT_LOCATION,
         []<typename T>() { SNITCH_CHECK(1 == 2); });
 
     r.add_with_types<int, float>(
-        {"typed test with tags pass", "[tag1]"}, {__FILE__, __LINE__}, []<typename T>() {});
+        {"typed test with tags pass", "[tag1]"}, SNITCH_CURRENT_LOCATION, []<typename T>() {});
     r.add_with_types<int, float>(
-        {"typed test with tags fail", "[tag1]"}, {__FILE__, __LINE__},
+        {"typed test with tags fail", "[tag1]"}, SNITCH_CURRENT_LOCATION,
         []<typename T>() { SNITCH_CHECK(1 == 2); });
 
     r.add_fixture(
-        {"MyClass", "test fixture pass", "[tag with space]"}, {__FILE__, __LINE__}, []() {});
-    r.add_fixture({"MyClass", "test fixture fail", "[tag with space]"}, {__FILE__, __LINE__}, []() {
-        SNITCH_CHECK(1 == 2);
-    });
+        {"MyClass", "test fixture pass", "[tag with space]"}, SNITCH_CURRENT_LOCATION, []() {});
+    r.add_fixture(
+        {"MyClass", "test fixture fail", "[tag with space]"}, SNITCH_CURRENT_LOCATION,
+        []() { SNITCH_CHECK(1 == 2); });
 
-    r.add({"test SUCCEED pass"}, {__FILE__, __LINE__}, []() { SNITCH_SUCCEED("something good"); });
-    r.add({"test FAIL fail"}, {__FILE__, __LINE__}, []() { SNITCH_FAIL_CHECK("something bad"); });
-    r.add({"test expression pass"}, {__FILE__, __LINE__}, []() { SNITCH_CHECK(1 == 1); });
-    r.add({"test expression fail"}, {__FILE__, __LINE__}, []() { SNITCH_CHECK(1 == 2); });
-    r.add({"test long expression pass"}, {__FILE__, __LINE__}, []() {
+    r.add(
+        {"test SUCCEED pass"}, SNITCH_CURRENT_LOCATION, []() { SNITCH_SUCCEED("something good"); });
+    r.add(
+        {"test FAIL fail"}, SNITCH_CURRENT_LOCATION, []() { SNITCH_FAIL_CHECK("something bad"); });
+    r.add({"test expression pass"}, SNITCH_CURRENT_LOCATION, []() { SNITCH_CHECK(1 == 1); });
+    r.add({"test expression fail"}, SNITCH_CURRENT_LOCATION, []() { SNITCH_CHECK(1 == 2); });
+    r.add({"test long expression pass"}, SNITCH_CURRENT_LOCATION, []() {
         SNITCH_CHECK(
             some_very_long_name_that_forces_lines_to_wrap ==
             some_very_long_name_that_forces_lines_to_wrap);
     });
-    r.add({"test long expression fail"}, {__FILE__, __LINE__}, []() {
+    r.add({"test long expression fail"}, SNITCH_CURRENT_LOCATION, []() {
         SNITCH_CHECK(
             some_very_long_name_that_forces_lines_to_wrap !=
             some_very_long_name_that_forces_lines_to_wrap);
     });
-    r.add({"test too long expression pass"}, {__FILE__, __LINE__}, []() {
+    r.add({"test too long expression pass"}, SNITCH_CURRENT_LOCATION, []() {
         std::string super_long_string(2 * snitch::max_message_length, 'a');
         SNITCH_CHECK(super_long_string == super_long_string);
     });
-    r.add({"test too long expression fail"}, {__FILE__, __LINE__}, []() {
+    r.add({"test too long expression fail"}, SNITCH_CURRENT_LOCATION, []() {
         std::string super_long_string(2 * snitch::max_message_length, 'a');
         SNITCH_CHECK(super_long_string != super_long_string);
     });
-    r.add({"test too long message pass"}, {__FILE__, __LINE__}, []() {
+    r.add({"test too long message pass"}, SNITCH_CURRENT_LOCATION, []() {
         std::string super_long_string(2 * snitch::max_message_length, 'a');
         SNITCH_FAIL(super_long_string);
     });
-    r.add({"test too long message fail"}, {__FILE__, __LINE__}, []() {
+    r.add({"test too long message fail"}, SNITCH_CURRENT_LOCATION, []() {
         std::string super_long_string(2 * snitch::max_message_length, 'a');
         SNITCH_FAIL(super_long_string);
     });
 
 #if SNITCH_WITH_EXCEPTIONS
-    r.add({"test NOTHROW pass"}, {__FILE__, __LINE__}, []() {
+    r.add({"test NOTHROW pass"}, SNITCH_CURRENT_LOCATION, []() {
         SNITCH_CHECK_NOTHROW(throw_something(false));
     });
-    r.add({"test NOTHROW fail"}, {__FILE__, __LINE__}, []() {
+    r.add({"test NOTHROW fail"}, SNITCH_CURRENT_LOCATION, []() {
         SNITCH_CHECK_NOTHROW(throw_something(true));
     });
-    r.add({"test THROW pass"}, {__FILE__, __LINE__}, []() {
+    r.add({"test THROW pass"}, SNITCH_CURRENT_LOCATION, []() {
         SNITCH_CHECK_THROWS_MATCHES(
             throw_something(true), std::runtime_error,
             snitch::matchers::with_what_contains{"I threw"});
     });
-    r.add({"test THROW fail"}, {__FILE__, __LINE__}, []() {
+    r.add({"test THROW fail"}, SNITCH_CURRENT_LOCATION, []() {
         SNITCH_CHECK_THROWS_MATCHES(
             throw_something(false), std::runtime_error,
             snitch::matchers::with_what_contains{"I threw"});
@@ -108,19 +111,19 @@ void register_tests_for_reporters(snitch::registry& r) {
             snitch::matchers::with_what_contains{"I throws"});
     });
 
-    r.add({"test unexpected throw fail"}, {__FILE__, __LINE__}, []() {
+    r.add({"test unexpected throw fail"}, SNITCH_CURRENT_LOCATION, []() {
         throw std::runtime_error("unexpected error");
     });
 #endif
 
-    r.add({"test SKIP"}, {__FILE__, __LINE__}, []() { SNITCH_SKIP("not interesting"); });
+    r.add({"test SKIP"}, SNITCH_CURRENT_LOCATION, []() { SNITCH_SKIP("not interesting"); });
 
-    r.add({"test INFO"}, {__FILE__, __LINE__}, []() {
+    r.add({"test INFO"}, SNITCH_CURRENT_LOCATION, []() {
         SNITCH_INFO("info");
         SNITCH_FAIL_CHECK("failure");
     });
 
-    r.add({"test multiple INFO"}, {__FILE__, __LINE__}, []() {
+    r.add({"test multiple INFO"}, SNITCH_CURRENT_LOCATION, []() {
         SNITCH_FAIL_CHECK("failure 1");
         SNITCH_INFO("info 1");
         SNITCH_FAIL_CHECK("failure 2");
@@ -131,13 +134,13 @@ void register_tests_for_reporters(snitch::registry& r) {
         SNITCH_FAIL_CHECK("failure 4");
     });
 
-    r.add({"test SECTION"}, {__FILE__, __LINE__}, []() {
+    r.add({"test SECTION"}, SNITCH_CURRENT_LOCATION, []() {
         SNITCH_SECTION("section") {
             SNITCH_FAIL_CHECK("failure");
         }
     });
 
-    r.add({"test multiple SECTION"}, {__FILE__, __LINE__}, []() {
+    r.add({"test multiple SECTION"}, SNITCH_CURRENT_LOCATION, []() {
         SNITCH_SECTION("section 1") {
             SNITCH_FAIL_CHECK("failure 1");
         }
@@ -157,7 +160,7 @@ void register_tests_for_reporters(snitch::registry& r) {
         SNITCH_FAIL_CHECK("failure 7");
     });
 
-    r.add({"test SECTION & INFO"}, {__FILE__, __LINE__}, []() {
+    r.add({"test SECTION & INFO"}, SNITCH_CURRENT_LOCATION, []() {
         SNITCH_INFO("info 1");
         SNITCH_SECTION("section 1") {
             SNITCH_INFO("info 2");

